crates/ruma-events/src/call/member.rs

```diff
@@ -0,0 +1,386 @@
+    /// An optional identifier for calls. Only relevant for some calls.
+    pub call_id: String,
+    /// Who owns/joins/controls (can modify) the call.
+    pub scope: CallScope,
+}
+
+/// The call scope defines different call ownership models.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+#[non_exhaustive]
+pub enum CallScope {
+    /// A call which every user of a room can join and create.
+    /// there is no particular name associated with it.
+    /// There can only be one per room.
+    #[serde(rename = "m.room")]
+    Room,
+    /// A user call is owned by a user. Each user can create one
+    /// there can be multiple per room. They are started and ended by
+    /// the owning user.
+    #[serde(rename = "m.user")]
+    User,
+}
+
+#[cfg(test)]
+mod tests {
+    use std::time::Duration;
+
+    use ruma_common::MilliSecondsSinceUnixEpoch as TS;
+    use serde_json::json;
+
+    use super::{
+        Application, CallApplicationContent, CallMemberEventContent, CallScope, Foci, LivekitFoci,
+        Membership,
+    };
+
+    fn _remove_whitespace(s: &str) -> String {
+        s.chars().filter(|c| !c.is_whitespace()).collect::<String>()
+    }
+    fn create_call_member_event_content() -> CallMemberEventContent {
+        CallMemberEventContent::new(vec![Membership {
+            application: Application::Call(CallApplicationContent {
+                call_id: "123456".to_owned(),
+                scope: CallScope::Room,
+            }),
+            device_id: "ABCDE".to_owned(),
+            expires: 36000,
+            foci_active: vec![Foci::Livekit(LivekitFoci {
+                alias: "1".to_owned(),
+                service_url: "https://livekit.com".to_owned(),
+            })],
+            membership_id: "0".to_owned(),
+            created_ts: None,
+        }])
+    }
+
+    #[test]
+    fn serialize_call_member_event_content() {
+        let call_member_event = &json!({
+            "memberships": [
+                {
+                    "application": "m.call",
+                    "call_id": "123456",
+                    "scope": "m.room",
+                    "device_id": "ABCDE",
+                    "expires": 36000,
+                    "foci_active": [
+                        {
+                            "livekit_alias": "1",
+                            "livekit_service_url": "https://livekit.com",
+                            "type": "livekit"
+                        }
+                    ],
+                    "membershipID": "0"
+                }
+            ]
+        });
+        println!("{}", serde_json::to_string_pretty(&create_call_member_event_content()).unwrap());
+
+        assert_eq!(
+            call_member_event,
+            &serde_json::to_value(create_call_member_event_content()).unwrap()
+        );
+    }
+
+    #[test]
+    fn deserialize_call_member_event_content() {
+        fn create_call_member_event_content() -> CallMemberEventContent {
```

Comment:
Why do you declare a method here? If it's called only once, you should just declare a variable.